Fixed parsing errors to show semantic analysis errors

TBolton2000 opened the pull request:
At least my compiler doesn't allow you to proceed to CFG generation and semantic analysis unless there are no semantic or parsing errors. I have fixed this program to remove the parsing errors, so I can show the semantic error of function redefinition.

khanhtn1 reviewed (changes requested):
; after funcBody is required. By deleting them you are introducing parsing errors :)

I would prefer that you refactor the main() out and create another error case.

TBolton2000 commented:
Good catch, I completely missed the semicolon being required after function declarations, but have fixed that in my compiler. I have reverted the changes on error003.txt and changed the comment to say what parse errors are expected, and added error008.txt for the semantic error function redefinition.
